tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,93 @@
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server.Query("CREATE DATABASE $database")
+        $server.Query("CREATE DATABASE $database2")
+
+    }
+    AfterAll {
+        $null = Get-DbaDatabase -SqlInstance $server -Databases @($database, $database2) | Remove-DbaDatabase -Confirm:$false
+        Remove-Item $script:PesterOutputPath -recurse -erroraction silentlycontinue  #clear test folder contents
+
+    }
+    AfterEach {
+        Remove-Item $script:PesterOutputPath -Recurse -erroraction silentlycontinue #clear test folder contents, doesn't need to fail if cleanup fails
+    }
+
+
+    Context "verifying output when running queries" {
         It "runs a specific query" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage' *>&1
-            $results.Name.Count | Should Be 1
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage'
+            @($results).Count | Should -Be 1
         }
         It "works with DatabaseSpecific" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific *>&1
-            $results.Name.Count -gt 10 | Should Be $true
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific
+            @($results).count | Should -BeGreaterThan 10
         }
     }
-}
\ No newline at end of file
+
+    context "verifying output when exporting queries as files instead of running" {
+
+        It "exports queries to sql files without running" {
+            Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -ExportQueries -QueryName 'Memory Clerk Usage' -OutputPath $script:PesterOutputPath
+            @(Get-ChildItem -path $script:PesterOutputPath -filter *.sql).Count | Should -be 1
+        }
+
+        It "returns pscustomobject[] containing parsed queries" {
+            [System.Management.Automation.PSObject[]]$results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -whatif
+            @($results).count | Should -BeGreaterThan 10
+
+            #verifying the data types returned are consistent with an array of pscustom customobjects to allow working with the queries further if needed
+            #this also ensures that any "whatif" is not polluting the output stream with objects
+            # for better understanding on why I'm using write-output to validate  type, see: https://github.com/pester/Pester/issues/38
+
+            write-debug "Note that to match , results have to be declared as above, explicitly psobject, otherwise just Object[] array"
+            Write-Output -NoEnumerate $results | Should -BeOfType [System.Management.Automation.PSObject[]]
+            if ($DebugPreference -ne 'silentlycontinue') {  $results | foreach-object { write-debug "foreach-object in `$results - UnderlyingSystemType $($_.GetType().UnderlyingSystemType)" } }
```

Comment:
remove the entire line

Reply:
will do, but see comment above please, as had reason for doing, to ensure pure `[pscustomobject[]]` returned.

Reply:
comment was on line 57. we don't need CI to run any debug. debugging is for humans hunting bugs, not computers.